src/MessageHandler/DeleteFileHandler.php

```diff
@@ -45,13 +56,33 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
      */
     public function __invoke(DeleteFile $deleteFile)
     {
-        $filePath = $deleteFile->getFilePath();
-        $this->logger->info(sprintf('Processing File with ID: "%s"', $filePath));
-        try {
-            $this->datastore->deleteFile($filePath);
-        } catch (\Exception $e) {
-            $this->logger->error(sprintf('Unable to delete file. Message: "%s"', $e->getMessage()));
+        $datasetSubmissionId = $deleteFile->getDatasetSubmissionId();
+
+        $this->logger->info(sprintf('Processing File with ID: "%s"', $datasetSubmissionId));
+
+        $datasetSubmission = $this->entityManager->getRepository(DatasetSubmission::class)->find($datasetSubmissionId);
+
+        if (!$datasetSubmission instanceof DatasetSubmission) {
+            $this->logger->warning('Submission not found');
             return;
         }
+
+        $fileset = $datasetSubmission->getFileset();
+
+        if (!$fileset instanceof Fileset) {
+            $this->logger->warning('fileset not found');
+            return;
+        }
+
+        foreach ($fileset->getDeletedFiles() as $deletedFile) {
+            try {
+                $this->datastore->deleteFile($deletedFile->getPhysicalFilePath());
+            } catch (\Exception $e) {
+                $this->logger->error(sprintf('Unable to delete file. Message: "%s"', $e->getMessage()));
+            }
+            $fileset->removeFile($deletedFile);
+        }
+
+        $this->entityManager->flush($datasetSubmission);
```

Comment:
Do we need this to be passed?


From the Doctrine/ORM/EntityManager.php, flush() method comment 

>      If an entity is explicitly passed to this method only this entity and the cascade-persist semantics + scheduled inserts/removals are synchronized.